.github/workflows/release.yaml

```diff
@@ -0,0 +1,304 @@
+          context: .
+          file: package/Dockerfile.suc
+          push: true
+          tags: ${{ env.IMAGE }}:${{ env.TAG_SUC }}
+          platforms: ${{ matrix.os }}/${{ matrix.arch }}
+
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v4
+        with:
+          name: "suc-digests-${{ matrix.os }}-${{ matrix.arch }}"
+          path: /tmp/digests/*
+          if-no-files-found: error
+          overwrite: true
+
+
+  merge-suc:
+    runs-on: ubuntu-latest
+    needs:
+      - build-push-image-suc
+    permissions:
+      contents: read
+      id-token: write # needed for the Vault authentication
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Load secrets from Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Download digests
+        uses: actions/download-artifact@v4
+        with:
+          path: /tmp/digests
+          pattern: suc-digests-*
+          merge-multiple: true
+
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          docker buildx imagetools create --tag ${{ env.IMAGE }}:${{ env.TAG_SUC }} $(printf '${{ env.IMAGE }}@sha256:%s ' *)
+
+      - name: Inspect image
+        run: |
+          docker buildx imagetools inspect  ${{ env.IMAGE }}:${{ env.TAG_SUC }}            
+
+  github_release:
+    runs-on: ubuntu-latest
+    needs:
+      - merge
+      - merge-suc
+    permissions:
+      contents: write # needed for creating the GH release
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download binary
+        uses: actions/download-artifact@v4
+        with:
+          pattern: binary-*
+          path: bin
+
+      - name: Prepare release
+        run: |
+          mkdir tmp
+          for DIR in bin/binary-*; do
+            if [[ -d "$DIR" && -f "$DIR/rancher-system-agent" ]]; then
+              ARCH=$(echo "$DIR" | awk -F'-' '{print $NF}')
+              mv "$DIR/rancher-system-agent" "tmp/rancher-system-agent-$ARCH"
+            fi
+          done
+          cp install.sh system-agent-uninstall.sh tmp
+          mkdir dist
+          for file in tmp/*; do
+            sha256sum "$file" >> dist/sha256sum.txt
```

Comment:
nit: because we're adding sums for all files into a single text file the name should probably be `sha256sums.txt`

Reply:
I initially thought sha256sum was the command name, so I didn't include the 's'. However, I understand your points now. To minimize the impact on the PR review, I will leave it as it is.